src/tools/westcos-tool.c

```diff
@@ -678,6 +678,10 @@ int main(int argc, char *argv[])
 		}
 
 		r = sc_pkcs15_encode_pubkey(ctx, &key, &pdata, &lg);
+		if(dst->modulus.data)
```

Comment:
On line 582, the `key` structure is initialized to zeroes so this condition is bogus. Calling just free should be enough. In the following condition too.